include/mgmt/fmfu_mgmt.h

```diff
@@ -24,15 +24,35 @@
 extern "C" {
 #endif
 
+enum fmfu_evt_id {
+	/** Prepare for full modem update to start. */
+	FMFU_EVT_INIT,
+	/** Full modem update finished. */
+	FMFU_EVT_FINISHED,
+	/** Full modem update failed. */
+	FMFU_EVT_ERROR,
+};
+
+/**
+ * @brief FMFU event callback function.
+ *
+ * @param event_id Event ID.
+ *
+ */
+typedef void (*fmfu_callback_t)(enum fmfu_evt_id id);
+
 /** @brief Setup and register the command handler for full modem
  *  update command handler group.
  *
- * The modem library must be initialized in DFU mode before calling this
- * function - nrf_modem_lib_init(FULL_DFU_MODE).
+ * The modem library must be initialized in DFU mode when a FMFU_EVT_INIT event
```

Comment:
Shouldn't this be part of the callback documentation? Perhaps list the possible events and state what handling is needed (or only the one event where you do need special handling)

Reply:
Another alternative is to have the docs in the event enum itself

Reply:
Sure I'll add a bit more to the documentation